app/controllers/katello/api/v2/repositories_controller.rb

```diff
@@ -254,23 +258,7 @@ def export
     end
 
     api :PUT, "/repositories/:id", N_("Update a repository")
-    param :name, String, :desc => N_("New name for the repository")
-    param :id, :number, :required => true, :desc => N_("repository ID")
```

Comment:
This should work fine once the `id` param is added to the :repo param group :)

Reply:
[vagrant@centos7-hammer-devel ~]$ hammer repository update --id=2 --name=Zoo1
Could not update the repository:
  Error: Unrecognised option '--id'.

Reply:
doh, good catch!